added dev-python/influxdb-client with deps

DefauIt opened the pull request:
Dependencies: ciso8601 randomize
Docs: https://influxdb-client.readthedocs.io/en/latest/

InfluxDB 2.0 python client library.
Note: Use this client library with InfluxDB 2.x and InfluxDB 1.8+.

DefauIt commented:
Closes: https://bugs.gentoo.org/782091
Package-Manager: Portage-3.0.17, Repoman-3.0.2
Signed-off-by: Ruslan Kh. root@xakru.me

00-matt commented on dev-python/ciso8601/ciso8601-2.1.3.ebuild:
```diff
@@ -0,0 +1,30 @@
+# Copyright 1999-2020 Gentoo Authors
```
```suggestion
# Copyright 1999-2021 Gentoo Authors
```

00-matt commented on dev-python/ciso8601/ciso8601-2.1.3.ebuild:
```diff
@@ -0,0 +1,30 @@
+KEYWORDS="~amd64 ~arm ~arm64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="test"
+
+RDEPEND=""
```
You can drop this if it's empty.

00-matt commented on dev-python/ciso8601/ciso8601-2.1.3.ebuild:
```diff
@@ -0,0 +1,30 @@
+python_test() {
+	nosetests --verbose || die
+	py.test -v -v || die
+}
```
Files should end with an empty line.

00-matt commented on dev-python/ciso8601/metadata.xml:
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="person">
+    <email>root@gentoo.org</email>
```
Is this really your email address? :)

You should mark yourself as proxied and add the proxy-maint project too:

```xml
  <maintainer type="person" proxied="yes">
    <email>you@your-real-email.com</email>
    <name>Ruslan Kh.name>
  </maintainer>
  <maintainer type="project" proxied="proxy">
    <email>proxy-maint@gentoo.org</email>
    <name>Proxy Maintainers</name>
  </maintainer>
```

00-matt commented on dev-python/influxdb-client/metadata.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="person">
```
Same as previous comment on metadata.

00-matt commented on dev-python/randomize/metadata.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="person">
```
Same as previous comment on metadata.

00-matt commented on dev-python/randomize/randomize-0.14.ebuild:
```diff
@@ -0,0 +1,22 @@
+# Copyright 1999-2020 Gentoo Authors
```
```suggestion
# Copyright 1999-2021Gentoo Authors
```

scardracs replied:
Leave the space ;)

00-matt commented on dev-python/randomize/randomize-0.14.ebuild:
```diff
@@ -0,0 +1,22 @@
+S="${WORKDIR}/nose-${PN}-master"
+
+
+LICENSE="GNU LGPL"
```
Which version of the LGPL? Also it should be a file in the [licenses directory](https://github.com/gentoo/gentoo/tree/master/licenses), e.g. `LGPL-3+`

00-matt commented on dev-python/randomize/randomize-0.14.ebuild:
```diff
@@ -0,0 +1,22 @@
+
+DEPEND="${REDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	"
```
Files should end with an empty line.

scardracs replied:
If you use .zip files in SRC_URI you must add app-arch/unzip or it will fails to unzip (if someone does not have unzip). I suggest to use .tar.gz file version

00-matt commented on dev-python/randomize/randomize-0.14.ebuild:
```diff
@@ -0,0 +1,22 @@
+
+DESCRIPTION="Randomize the order of the tests within a unittest.TestCase class"
+HOMEPAGE="https://github.com/nloadholtes/nose-randomize"
+SRC_URI="https://github.com/nloadholtes/nose-randomize/archive/master.zip"
```
```suggestion
SRC_URI="https://github.com/nloadholtes/nose-randomize/archive/master.zip -> ${P}.zip"
```

scardracs replied:
SRC_URI="https://github.com/nloadholtes/nose-randomize/archive/master.tar.gz -> ${P}.tar.gz"

00-matt commented:
Also need to split your commits up [according to the Gentoo git workflow](https://wiki.gentoo.org/wiki/Gentoo_git_workflow#Commit_policy). Feel free to join [#gentoo-proxy-maint on freenode](https://webchat.freenode.net/?channels=#gentoo-proxy-maint) if you need any help.

scardracs commented on dev-python/ciso8601/ciso8601-2.1.3.ebuild:
```diff
@@ -0,0 +1,30 @@
+IUSE="test"
+
+RDEPEND=""
+DEPEND="${REDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? (
+		dev-python/nose[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+	)"
```
If you use .zip files in SRC_URI you must add app-arch/unzip or it will fails to unzip (if someone does not have unzip). I suggest to use .tar.gz file version

scardracs commented on dev-python/randomize/randomize-0.14.ebuild:
```diff
@@ -0,0 +1,22 @@
+
+EAPI="7"
+
+PYTHON_COMPAT=( python3_{4,7,8} )
```
python3_4 is not used anymore. New packages must be tested at least against 7, 8 and 9. Same thing for other ebuilds

scardracs commented on dev-python/influxdb-client/influxdb-client-1.16.0.ebuild:
```diff
@@ -0,0 +1,46 @@
+	ciso? (
+		>=dev-python/ciso8601-2.1.1
+	)
+"
```
Always leave a newline on every ebuilds

scardracs commented on dev-python/ciso8601/metadata.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="person">
```
<maintainer type="person" proxied="yes">

scardracs commented on dev-python/ciso8601/metadata.xml:
```diff
@@ -0,0 +1,13 @@
+    <email>root@xakru.me</email>
+    <name>Ruslan Kh.</name>
+  </maintainer>
+  <stabilize-allarches/>
```
Be sure to add proxi-maint as you are not a full dev (with @gentoo.org mail)

scardracs commented on dev-python/ciso8601/Manifest:
```diff
@@ -0,0 +1,3 @@
+DIST ciso8601-2.1.3.tar.gz 15726 BLAKE2B 2122c6b677ef0386bfab30fb76d746ff0d22e402464a3e0b928b71d0c22cf9dc26503dba52750015f45b4397ebd1dca16400873d2f67ce38dac2b5fbed2402dc SHA512 9342cf9b2ded1077a2c3906527ae4ac05dfc0e8104f1c85e041d636caa1180df390f01b2d87a33a4c197f4467a25bda1cd0136eb48585b127662cde7d0ffda7e
+EBUILD ciso8601-2.1.3.ebuild 671 BLAKE2B d3b7585c062217a25f1c68c9b7aeea07ce0099a850ddc35e159c6db29aa884efc2f68bae1b226b8546f7fc848dec52f9415f925326ae557b293a161d6d135208 SHA512 811bb77f6d682bb0bae96b09bbe59ccdd2275790bf5aa67493b67652dc79438893a74699d65e1e07da41199be0603531f6812ae2370fd2052eda860a36198396
+MISC metadata.xml 404 BLAKE2B 19edddec02504a5f05ac1a5ce6dba6faec8031c4a82d3c3ed380a478157213bb6bcd80ae9410c8f92bb2f76bd784d207ba41917471d4781f5f5c2ace48f169c0 SHA512 d3541c13ee1f7a79e3b096bb19d4a740792938de1752784387d6e6f8a31f166a216cdb60dbfc2cf68121c0c9979e5839a91a647411ce5fb831b93ead8afbe98f
```
Your metadatas seems wrong... Have you run ebuild **packagename.ebuild** manifest

scardracs replied:
This comment is for every metadata you have pushed

scardracs replied:
You have to run ebuild only with .ebuild file, not with metadata.xml

scardracs commented on dev-python/ciso8601/ciso8601-2.1.3.ebuild:
```diff
@@ -0,0 +1,28 @@
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="test"
```
If you add IUSE="test" then you must have to add RESTRICT="!test? ( test )" because we don't want to run test every time but just when it is needed

scardracs replied:
This comment is for every ebuilds you have made

scardracs commented on dev-python/influxdb-client/influxdb-client-1.16.0.ebuild:
```diff
@@ -0,0 +1,47 @@
+		>=dev-python/nose-1.3.7[${PYTHON_USEDEP}]
+		>=dev-python/pluggy-0.3.1[${PYTHON_USEDEP}]
+		>=dev-python/py-1.4.31[${PYTHON_USEDEP}]
+		>=dev-python/randomize-0.13[${PYTHON_USEDEP}]
```
We don't have such package. If you need it then you must PR it too

DefauIt replied:
in portage i can't find only dev-python/randomize
should i create new PR for this package?

scardracs replied:
Does dev-python/randomize actually exist as package? If yes you have to port it (if an ebuild exist and it isn't made by you please ask to maintainer if you could take it)

scardracs commented on dev-python/influxdb-client/influxdb-client-1.16.0.ebuild:
```diff
@@ -0,0 +1,47 @@
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="ciso extra test"
```
Ciso and extra are not know USE flags. Be sure to add them on your metadata.xml after maintainer. Follow my example if you want https://dpaste.com/HRJG4GCAH

scardracs commented on dev-python/influxdb-client/influxdb-client-1.16.0.ebuild:
```diff
@@ -0,0 +1,47 @@
+
+DOCS="README.rst"
+
+RDEPEND=">=dev-python/Rx-3.0.1[${PYTHON_USEDEP}]
+	>=dev-python/certifi-14.05.14[${PYTHON_USEDEP}]
+	>=dev-python/six-1.10[${PYTHON_USEDEP}]
+	>=dev-python/python-dateutil-2.5.3[${PYTHON_USEDEP}]
+	>=dev-python/urllib3-1.15.1[${PYTHON_USEDEP}]
+	>=dev-python/pytz-2019.1[${PYTHON_USEDEP}]
```
Please order them alphabetically

scardracs commented on dev-python/influxdb-client/influxdb-client-1.16.0.ebuild:
```diff
@@ -0,0 +1,47 @@
+"
+BDEPEND="${REDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	extra? (
+		>=dev-python/pandas-0.25.3
+		dev-python/numpy
+	)
+	test? (
+		>=dev-python/coverage-4.0.3[${PYTHON_USEDEP}]
+		>=dev-python/nose-1.3.7[${PYTHON_USEDEP}]
+		>=dev-python/pluggy-0.3.1[${PYTHON_USEDEP}]
+		>=dev-python/py-1.4.31[${PYTHON_USEDEP}]
+		>=dev-python/randomize-0.13[${PYTHON_USEDEP}]
+		>=dev-python/pytest-5.0.0[${PYTHON_USEDEP}]
+		>=dev-python/httpretty-1.0.2[${PYTHON_USEDEP}]
+	)
+	ciso? (
+		>=dev-python/ciso8601-2.1.1
+	)
```
Order alphabetically. The ? flags should be at the bottom (as you do)

scardracs commented on dev-python/influxdb-client/metadata.xml:
```diff
@@ -0,0 +1,13 @@
+    <email>root@xakru.me</email>
+    <name>Ruslan Kh.</name>
+  </maintainer>
+  <stabilize-allarches/>
```
Are you sure it works for every arches?

scardracs commented:
I haven't checked every files, but please fix suggestions made by me and @00-matt. Also please come on freenode #gentoo-maint if you need more help

scardracs commented on dev-python/influxdb-client/Manifest:
```diff
@@ -0,0 +1,3 @@
+DIST influxdb-client-1.16.0.tar.gz 1587901 BLAKE2B dab5dd6b2349b61101f69ff30ee9a087eb383361b07797e5e49c51e29abcda3e26fbbe63ea2880932d74979435b34e2adb5093057a17a79f506918324a4db497 SHA512 1051b3929a933a7879755ef8894efd19831f361c895ce3dcd6f9747753ef09b9446503f45cb0f0fd153c95ebd0679bad9ac54be2581b9644c17cba33774fee0b
+EBUILD influxdb-client-1.16.0.ebuild 1311 BLAKE2B f49b3d8b53db3cf7b4686d3caadd838119866bd6087d8db37ccf398048a74bb8a7360aa467b82289a1445354067c7c5caa712f85d92c3548ebf00359433feeb5 SHA512 b1765bb5017793ca0fa5a257645e8625cd1ad818093e018c303bb1ec8ac7e3eb1527f39c3451cf3fd09f9e9bd49fb652c34a01d2d82c274d06e2502b74585fe4
+MISC metadata.xml 428 BLAKE2B d60f8d9461546273883f3bbcd3c31af90a2b66ff62f88652970fe8a33124c335530d49522c0e41a2fde46010a5fa9ba352b746a2c7878daee1a7de623b172f37 SHA512 07365d1f99c6a64aae6d24eb41f3b7c339e9aa79027b4b5ffda49646ca690fa44f4c24e9b4817359800ca14fe3381723471b0efac6ff6f0b35d8b6fe27490296
```
Same as before

scardracs commented on dev-python/ciso8601/ciso8601-2.1.3.ebuild:
```diff
@@ -0,0 +1,28 @@
+
+EAPI="7"
+
+PYTHON_COMPAT=( python3_{7,8} )
```
If it possible use PYTHON_COMPAT=( python3_{7..9} ) if it works with 9 too because now packages should works with it too and we prefer to test them against latest version of python

DefauIt replied:
I'll try to build with Python-3.9, but i'm not maintain ciso8601 sources.
and https://pypi.org/project/ciso8601/ - python3.9 not listed in supported versions

scardracs replied:
Try it, if doesn't work try to figure out why and if it an upstream problem leave a comment

scardracs commented on dev-python/ciso8601/ciso8601-2.1.3.ebuild:
```diff
@@ -0,0 +1,28 @@
+
+DESCRIPTION="Fast ISO8601 date time parser for Python written in C"
+HOMEPAGE="https://github.com/closeio/ciso8601 https://pypi.org/project/ciso8601/"
+SRC_URI="mirror://pypi/${P:0:1}/${PN}/${P}.tar.gz"
```
We don't really like mirror:// links. If possible use a full link like SRC_URI="https://github.com/closeio/ciso8601/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"

DefauIt replied:
thank you for the detailed explanation

scardracs replied:
You are welcome :)

scardracs commented on dev-python/randomize/Manifest:
```diff
@@ -0,0 +1,3 @@
+DIST randomize-0.14.tar.gz 5267 BLAKE2B 2151f5af761b63bdf8e385b5c541b4ed2806961bc7c302d17696ee8a21765bb30924603a7b93cf6ef07747a0719078589dd89b832767825b6481cfd7672b6ba2 SHA512 d168fb4a45026c75415449c661e131aabc5a9533346fb07d3cf2f46aa7e21b2ed1cca13077fc650ffd7609c27edad1eeeafb67bc0bbab974c59eb5bf22ec8439
+EBUILD randomize-0.14.ebuild 563 BLAKE2B e82d3f4e4184ae3ceebb6ffef13461fa8d3b06012075e4d109e67a022575ab96f442b51902e26b1c8462010591231c5e53ecbbb8022b8b20d7693148d091e983 SHA512 9e27fa5d8a4ff7585a5af32b5b27d7c8e1763377dd62c0ac6713a57690ace6af4424584ff2129ffc8153912e34310f7f73e30d7039bd1058c4e5b3977ac6460d
+MISC metadata.xml 415 BLAKE2B 0b48c4b42c4aff7629c0e0e9a67f96e0ed17d63b3165c45540770f7acfe76b51dc1752ddc561fe2af8db4ad0cd2c8c6527daea6b70db2ce8d6f802de8f99c3a7 SHA512 381c42e3bc7d3f64617f4fa8b0a91576275cb6c9ddeeac2151e94f7fe7baa7421a623e1bbdffa47dd2e8c8f7e7c9c6e95371916ee31f42676d4ee823226b0cd5
```
Same as before

scardracs commented on dev-python/randomize/randomize-0.14.ebuild:
```diff
@@ -0,0 +1,19 @@
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+
+DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]"
```
Leave a newline

scardracs replied:
Remember to leave a newline on every ebuilds

scardracs reviewed (changes requested):
Please follows these suggestions :)

DefauIt commented:
Signed-off-by: Ruslan Khakimov root@xakru.me

scardracs commented on dev-python/ciso8601/metadata.xml:
```diff
@@ -0,0 +1,20 @@
+    <name>Proxy Maintainers</name>
+  </maintainer>
+  <use>
+    <flag name=test>Enable test support</flag>
```
"test"

DefauIt replied:
oops, missed. thanks again, fixed :)

scardracs commented:
Now you have just to remove EBUILD and METADATA from Manifest and you are good to go

DefauIt commented:
it was bad idea to replace tab indents by space indents, i'll fix it in a dew minutes.

scardracs commented:
@DefauIt only longdescription exist

DefauIt commented:
looked at https://devmanual.gentoo.org/ebuild-writing/misc-files/metadata/ for <description>

scardracs commented:
> looked at https://devmanual.gentoo.org/ebuild-writing/misc-files/metadata/ for `<description>`
> 

These are 2 different things :)

scardracs commented:
Ah, yes, longdescription mush have a minimum amount of characters

DefauIt commented:
updated text in `<longdescription>`, added more information.
can't find info about `<longdescription>` minimum amount of characters.

scardracs commented on dev-python/ciso8601/ciso8601-2.1.3.ebuild:
```diff
@@ -0,0 +1,30 @@
+	py.test -v -v || die
+}
+
+RESTRICT="!test? ( test )"
```
Put it after IUSE

scardracs commented on dev-python/influxdb-client/influxdb-client-1.16.0.ebuild:
```diff
@@ -0,0 +1,49 @@
+	)
+"
+
+RESTRICT="!test? ( test )"
```
Same as before

scardracs commented on dev-python/randomize/randomize-0.14.ebuild:
```diff
@@ -0,0 +1,19 @@
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+
+DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]"
```
Remember to leave a newline on every ebuilds

DefauIt replied:
hm... in i have 20 line in this file
```
wc -l randomize-0.14.ebuild
      20 randomize-0.14.ebuild
```

DefauIt replied:
ooops, it was \n, not a new empty line

scardracs replied:
Yeah it should be an empty line :)

DefauIt commented:
`git rebase -i master` - done successful, 
You’ve been very kind, and I appreciate your help!

vaukai commented on dev-python/ciso8601/metadata.xml:
```diff
@@ -17,9 +20,6 @@
     https://github.com/closeio/ciso8601#supported-subset-of-iso-8601
   </longdescription>
-  <use>
-    <flag name="test">Enable test support</flag>
```
Why not use the global "test" USE flag?

DefauIt replied:
of course, good idea :) thanks!

vaukai replied:
repoman or pkgcheck should have told you :-)

DefauIt commented:
I’m sorry for my long absence, I was sick.
Given PR cannot update,  can't build ciso8601 with Python-3.9.
ciso8601 currently supports up to 3.8 Python, https://github.com/closeio/ciso8601

DefauIt commented:
next week I’m going to build.... please do not close PR.

DefauIt commented:
- updated packages with their version
- added EAPI8
- dropped python3.7 support

can't test with python3.10, need dev-libs/libxml2 with python3.10 support

00-matt commented:
(If you want to please the QA bot you'll need to rebase on latest master so that distutils-r1 supports EAPI-8.)
